samza-sql/src/main/java/org/apache/samza/sql/data/SamzaSqlRelMessage.java

```diff
@@ -42,6 +42,14 @@
   private final List<String> names = new ArrayList<>();
   private final Object key;
 
+  public SamzaSqlRelMessage(List<String> names, List<Object> values) {
+    Validate.isTrue(names.size() == values.size(), "Field Names and values are not of same length.");
```

Comment:
Can you just call the other constructor passing the key as null?
That way we remove some duplicate code.